### PR TITLE
chore: cleanup some devtool console warnings

### PR DESCRIPTION
**Pull request by joshuaeilers**
## Checklist

Had some free time and thought I could clean up some of our js console / eslint warnings, make it a bit cleaner. A minimal amount of refactoring. To keep the PR small I'm only tackling one of the dropdowns.

## Home page

Before

Screenshot 2023-05-08 at 1 36 03 PM

Screenshot 2023-05-08 at 1 35 34 PM

After

Screenshot 2023-05-08 at 1 37 44 PM

Screenshot 2023-05-08 at 1 29 33 PM

## Member groups dropdown (should be visually unchaged)

Screenshot 2023-05-08 at 1 33 24 PM

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: datahub-web-react/package.json [hunk 1]
```diff
@@ -67,7 +67,7 @@
         "react-color": "^2.19.3",
         "react-dom": "^17.0.0",
         "react-email-share-link": "^1.0.3",
-        "react-helmet": "^6.1.0",
+        "react-helmet-async": "^1.3.0",
```
Comment (joshuaeilers) [inline datahub-web-react/package.json:70]: This upgrade resolves a js console warning and the old package react-helmet recommends upgrading to react-helmet-async.

File: datahub-web-react/src/App.tsx [hunk 2]
```diff
@@ -118,18 +118,20 @@ const App: React.VFC = () => {
...
-                    <title>{dynamicThemeConfig.content.title}</title>
-                </Helmet>
-                <EntityRegistryContext.Provider value={entityRegistry}>
-                    <ApolloProvider client={client}>
-                        <Routes />
-                    </ApolloProvider>
-                </EntityRegistryContext.Provider>
-            </Router>
-        </ThemeProvider>
+        <HelmetProvider>
```
Comment (joshuaeilers) [inline datahub-web-react/src/App.tsx:121]: Only change is this new provider.

File: datahub-web-react/src/app/search/autoComplete/ParentContainers.tsx [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import { FolderOpenOutlined } from '@ant-design/icons';
 import { Typography } from 'antd';
-import React from 'react';
+import React, { Fragment } from 'react';
```
Comment (joshuaeilers) [inline datahub-web-react/src/app/search/autoComplete/ParentContainers.tsx:3]: Refactored this component a bit so I could give both sets of React fragments unique keys from the urn's.

File: datahub-web-react/src/utils/test-utils/TestPageContainer.tsx [hunk 4]
```diff
@@ -53,33 +54,35 @@ export default ({ children, initialEntries }: Props) => {
...
-                                refetchCenterNode: () => {},
-                            }}
-                        >
-                            {children}
-                        </LineageExplorerContext.Provider>
-                    </UserContextProvider>
-                </EntityRegistryContext.Provider>
-            </MemoryRouter>
-        </ThemeProvider>
+        <HelmetProvider>
```
Comment (joshuaeilers) [inline datahub-web-react/src/utils/test-utils/TestPageContainer.tsx:57]: Only change here is added the HelmetProvider

File: datahub-web-react/src/app/entity/dataset/profile/stats/stats/DatasetStatsSummarySubHeader.tsx [hunk 5]
```diff
@@ -13,10 +9,11 @@ export const DatasetStatsSummarySubHeader = () => {
     const dataset = result?.dataset;
 
     const maybeStatsSummary = dataset?.statsSummary as DatasetStatsSummaryObj;
+
     const maybeLastProfile =
-        ((dataset?.datasetProfiles?.length || 0) > 0 && (dataset?.datasetProfiles![0] as DatasetProfile)) || undefined;
-    const maybeLastOperation =
-        ((dataset?.operations?.length || 0) > 0 && (dataset?.operations![0] as Operation)) || undefined;
+        dataset?.datasetProfiles && dataset.datasetProfiles.length ? dataset.datasetProfiles[0] : undefined;
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/entity/dataset/profile/stats/stats/DatasetStatsSummarySubHeader.tsx:14]: nice!

File: datahub-web-react/src/app/entity/group/GroupMembers.tsx [hunk 6]
```diff
@@ -155,28 +155,28 @@ export default function GroupMembers({ urn, pageSize, isExternalGroup, onChangeM
...
-        if (e.key === 'remove') {
-            onRemoveMember(urnID);
-        }
-    };
-
-    const menu = (urnID) => {
-        return (
-            <Menu onClick={(e) => onMemberMenuClick(e, urnID)}>
-                <Menu.Item disabled key="make">
+    const getItems = (urnID: string): MenuProps['items'] => {
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/entity/group/GroupMembers.tsx:158]: nice refactor!

File: datahub-web-react/src/app/entity/shared/containers/profile/sidebar/Ownership/EditOwnersModal.tsx [hunk 7]
```diff
@@ -199,21 +199,22 @@ export const EditOwnersModal = ({
...
-        // eslint-disable-next-line react/prop-types
-        const { label, closable, onClose } = props;
-        const onPreventMouseDown = (event) => {
-            event.preventDefault();
-            event.stopPropagation();
-        };
+    const tagRender = ({ closable, label, onClose }: { closable: boolean; label: ReactNode; onClose: () => void }) => {
         return (
-            <StyleTag onMouseDown={onPreventMouseDown} closable={closable} onClose={onClose}>
+            <StyleTag
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/entity/shared/containers/profile/sidebar/Ownership/EditOwnersModal.tsx:204]: nice cleanup!

File: datahub-web-react/src/app/search/autoComplete/quickFilters/QuickFilter.tsx [hunk 8]
```diff
@@ -8,7 +8,7 @@ import { getQuickFilterDetails } from './utils';
 import { ANTD_GRAY } from '../../../entity/shared/constants';
 import analytics, { Event, EventType } from '../../../analytics';
 
-const QuickFilterWrapper = styled(Button)<{ isSelected: boolean }>`
+const QuickFilterWrapper = styled(Button)<{ selected: boolean }>`
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/search/autoComplete/quickFilters/QuickFilter.tsx:11]: curious about isSelected > selected.

Just for brevity? Or is this a consistency thing you noticed?

File: datahub-web-react/src/app/recommendations/renderer/component/TagSearchList.tsx [hunk 9]
```diff
@@ -54,8 +54,8 @@ export const TagSearchList = ({ content, onClick }: Props) => {
     return (
         <TagSearchListContainer>
             {tags.map((tag, index) => (
-                <TagContainer>
-                    <TagButton type="link" key={tag.urn} onClick={() => onClickTag(tag, index)}>
+                <TagContainer key={tag.urn}>
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/recommendations/renderer/component/TagSearchList.tsx:57]: interesting! why was this one required?

**Review by jjoyce0510 — APPROVED** [review]
overall this looks great to me. few small questions

File: datahub-web-react/src/app/search/autoComplete/quickFilters/QuickFilter.tsx (hunk 8 above)
Comment (joshuaeilers) [reply datahub-web-react/src/app/search/autoComplete/quickFilters/QuickFilter.tsx:11]: Would not have normally changed this but I was getting js console warnings about isSelected not being a valid name style for a dom node.

**Pull request merged**